### PR TITLE
feat: Migrate Fastly management for edX into Pulumi

### PR DESCRIPTION
**Pull request by blarghmatey**
This moves the logic for managing the Fastly distribution for Open edX installations
into Pulumi.

## Description

- Adds a service definition for Fastly into edX Pulumi app
- Adds support for templating the MFE regex fields in the Fastly distribution

## Motivation and Context

We have been managing the Fastly distribution via click-ops up to now. As we work on enabling more MFEs it will be easier to keep things aligned by managing the Fastly configuration as code.

## How Has This Been Tested?

Imported the state of the MITx CI environment into the relevant stack and previewed the proposed diff to ensure it was reasonable.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/ol_infrastructure/applications/edxapp/__main__.py [hunk 1]
```diff
@@ -1269,6 +1272,159 @@ def cloud_init_user_data_func(
...
+            "backends",
+            "cache_settings",
+            "conditions",
+            "dictionaries",
+            "domains",
+            "name",
+            "snippets",
+        ],
+    }
+fastly_mitxonline_qa = fastly.ServiceVcl(
```
Comment (shaidar) [inline src/ol_infrastructure/applications/edxapp/__main__.py:1295]: Shouldn't this be generic/templated?

File: src/ol_infrastructure/applications/edxapp/__main__.py [hunk 2]
```diff
@@ -1269,6 +1272,159 @@ def cloud_init_user_data_func(
...
+            statement=f'req.url.path ~ "{mfe_regex}"',
+            type="REQUEST",
+        ),
+    ],
+    dictionaries=[
+        fastly.ServiceVclDictionaryArgs(
+            name="marketing_redirects",
+        )
+    ],
+    domains=[
```
Comment (shaidar) [inline src/ol_infrastructure/applications/edxapp/__main__.py:1342]: Do we not want the domain name to be pulled from the stack config?

File: src/ol_infrastructure/applications/edxapp/__main__.py (hunk 1 above)
Comment (blarghmatey) [reply src/ol_infrastructure/applications/edxapp/__main__.py:1295]: Good catch, fixing now...

File: src/ol_infrastructure/applications/edxapp/__main__.py (hunk 2 above)
Comment (blarghmatey) [reply src/ol_infrastructure/applications/edxapp/__main__.py:1342]: They should come from the config, I'll fix that

**Comment by shaidar** [comment]
Looks good now. I didn't compare the fastly config to what's in place now as I'm assuming u ran an export to get all that. 👍

**Pull request merged**